src/fileformats/ocd_file_import.cpp

```diff
@@ -1836,28 +1846,11 @@ void OcdFileImport::setupPointSymbolPattern(PointSymbol* symbol, std::size_t dat
 
 QImage OcdFileImport::importSymbolIcon(const Ocd::IconV8& icon)
 {
-	static const QColor palette[16] = {
-	    {   0,   0,   0 },
-	    { 128,   0,   0 },
-	    { 0,   128,   0 },
-	    { 128, 128,   0 },
-	    {   0,   0, 128 },
-	    { 128,   0, 128 },
-	    {   0, 128, 128 },
-	    { 128, 128, 128 },
-	    { 192, 192, 192 },
-	    { 255,   0,   0 },
-	    {   0, 255,   0 },
-	    { 255, 255,   0 },
-	    {   0,   0, 255 },
-	    { 255,   0, 255 },
-	    {   0, 255, 255 },
-	    { 255, 255, 255 }
-	};
+	static const auto palette = Ocd::IconV8::palette<QColor>();
+	constexpr int icon_size = Ocd::IconV8::size();
 	
 	auto icon_bits = icon.bits;
-	constexpr int icon_size = 22;
-	QImage image(icon_size, icon_size, QImage::Format_ARGB32_Premultiplied);
+	auto image = QImage{icon_size, icon_size, QImage::Format_ARGB32_Premultiplied};
```

Comment:
`QImage image {icon_size, icon_size, QImage::Format_ARGB32_Premultiplied};` is the way to go (ES.23).